### ISSUE TITLE
Apply range/sliders/missing init warnings to applicable blocks

### ISSUE THREAD

**Issue by MatzElectronics** [parallaxinc/BlocklyProp#944]
The number value block is currently very simple.  By giving it a few features, we can make BlocklyProp a bit more intuitive/informative for students.  Many block inputs have set ranges or allowed values, but its a bit difficult to communicate to the user what those values are without cluttering the block (look at the S3 motor blocks if you need an example).

There is a function in the block definitions called .setWarningText() that gives a visual cue if something is not correct, and it can be clicked on for more info.  The block label could also mutate to show a range as well.  Finally, the FieldAngle input type could be altered/added to to become a slider, giving the user an intuitive way to understand what an input's range is.

In summary, features to add to the number block:
- take a range or set of values from the input it is connected to
- display ranges, when appropriate, as the text on the number block
- throw a warning for out of range or illegal values
- add a slider option to the FieldRange input type
- use code embedded in the input connections to set the features of the number block on the fly

**Comment by MatzElectronics** [comment]
Inital tests of functionality look good, now to apply it everywhere that it's appropriate.

**Comment by MatzElectronics** [comment]
I think we have this handled.  Closing.
